### DISCUSSION TITLE
Subdomain authorization

### DISCUSSION THREAD

**Discussion by dkornilove** [umputun/remark42#1093] [Q&A]
Hello,
I have site http**s**://example.com
I have a dev copy of it on the same server with 'dev' subdomain http://dev.example.com
It can be accessed from the internet but it has not ssl
Remark is configured for the backend url _https://example.com/remark_ and working perfectly within the main site
About six month ago it does work the same way on the dev copy without any additional configurations
but unfortunately authorization does not work on the dev copy now
When i hit sign in in the remark form with any third party oauth app being on the http://dev.example.com/blogpost1  it jumps to the new tab then closed and I can see _https://example.com/remark/api/v1/user?site=site GET 401_ error with "Unauthorized" response. When I login with email the process completed fine and looks like Im logged as an email user but when i try to submit a message it still responsing as for unathorized user.
What has been changed and how should I handle the same endpoint but for dev subdomain without ssl?

**Comment by paskal** [comment]
Could you please send me the link to the prod and dev site here or privately?

**Reply by paskal** [reply]
What I found so far: Chrome doesn't work, Firefox and Safari do work. Definitely tied to some security policies difference, as a start, I would suggest trying to set `PROXY_CORS=true` and see if it changes anything, or the problem is elsewhere.

**Reply by dkornilove** [reply]
> What I found so far: Chrome doesn't work, Firefox and Safari do work. Definitely tied to some security policies difference, as a start, I would suggest trying to set `PROXY_CORS=true` and see if it changes anything, or the problem is elsewhere.

When I set PROXY_CORS=true then this happens 
_Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://example.com/remark/web/embed.mjs._
while trying to open dev remark page http://dev.example.com/post1

I also want to note that I am using the lateset master and <script> stuff from the readme.
In addition to chrome, the edge also does not work. Yandex browser work as well as Firefox thow

**Reply by umputun** [reply]
setting `PROXY_CORS=true` means "disable internal CORS and delegate it to proxy". So, you are asking here to let you handle it. In other words, it turns off the internal CORS support to allow handling by the external proxy server.

**Reply by dkornilove** [reply]
@umputun what If I am unable to handle cors stuff manually?
Maybe I should create an issue from now?

**Reply by umputun** [reply]
I'm not sure what you mean. The goal of this parameter is to allow you to handle the CORS externally, see the [original issue for more details](https://github.com/umputun/remark42/issues/749)

If you are unable to handle cors on proxy level, just don't set this parameter.

**Reply by paskal** [reply]
It was my suggestion which turned out to be wrong, what I need to see is if we can set more relaxed cors settings. I’ll check it once more when I’ll be near the computer.

**Reply by paskal** [reply]
Definitely, something is off with Chrome security defaults compared with Firefox and Safari. It's either something to do with HTTP vs HTTPS or with another subdomain. @dkornilove, would you be so kind as to test the HTTPS subdomain to verify which of two it is?
